docs/source/examples/dataframe-hdf5.rst

```diff
@@ -0,0 +1,111 @@
+Dataframes from HDF5 files
+=========================
+
+This section provides working examples of ``dask.dataframe`` methods to read HDF5 files. HDF5 is a unique technology suite that makes possible the management of extremely large and complex data collections. To learn more about HDF5, visit the `HDF Group Tutorial page <https://www.hdfgroup.org/HDF5/whatishdf5.html>`_.  For an overview of ``dask.dataframe``, its limitations, scope, and use, see the `DataFrame Overview section <http://dask.pydata.org/en/latest/dataframe-overview.html>`_. 
+
+  **Important Note** -- `h5py <http://docs.h5py.org/en/latest/index.html>`_ and `PyTables <http://www.pytables.org/index.html>`_ are the two premiere libraries to work with HDF5 data in Python.   ``dask.dataframe``, `like python pandas <http://pandas.pydata.org/pandas-docs/stable/io.html#io-hdf5>`_, uses the the `PyTables <http://www.pytables.org/FAQ.html#what-is-pytables>`_ library to interface with the HDF5 binary data format. HDF5 files made using the h5py project may not read into a dask DataFrame. 
+
+  While there is `planned work to refactor PyTables to use h5py bindings to HDF5 and create 100% compatibility <https://groups.google.com/forum/#!msg/pytables-users/GBVkvGexSag/IwLjEtIfKIgJ>`_, there are differences in how the interfaces are implemented. `dask.dataframe` will have `nearly seamless compatibility with HDF5 files built in pandas <https://github.com/bokeh/bokeh/issues/1246>`_.  Here is more information on the differences between *h5py* and *PyTables* and work to improve compatibility:
+
+    -  `h5py explanation of differences from PyTables <http://docs.h5py.org/en/latest/faq.html#what-s-the-difference-between-h5py-and-pytables>`_
+    -  `PyTables explanation of differences from h5py <http://www.pytables.org/FAQ.html#how-does-pytables-compare-with-the-h5py-project>`_
+    -  `HackFest with h5py and PyTables teams <https://curtinic.github.io/python-and-hdf5-hackfest/>`_
```

Comment:
While the h5py/pytables differences are of importance here they may also be a bit out of scope?  

I wonder if we should say "dask.dataframe.read_hdf is built on pandas.read_hdf and so has the same limitations.  See here (some link) for more information."